### PR TITLE
 nix-pin: 0.1.2 -> 0.2.2

### PR DESCRIPTION
**Pull request by timbertson**
Also added an updateScript for automating updates

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
This broke nix tarball evaluation: https://hydra.nixos.org/build/73131644/nixlog/1

File: pkgs/tools/package-management/nix-pin/default.nix [hunk 1]
```diff
@@ -16,12 +16,24 @@ let self = stdenv.mkDerivation rec {
...
     cp -r bin share "$out"
   '';
-  passthru = {
-    callWithPins = path: args:
-      import "${self}/share/nix/call.nix" {
-        inherit pkgs path args;
-      };
-  };
+  passthru =
+    let api = import "${self}/share/nix/api.nix" { inherit pkgs; }; in
```
Comment (Mic92) [inline pkgs/tools/package-management/nix-pin/default.nix]: This is probably not allowed in hydra?

**Comment by Mic92** [comment]
I have reverted the update for now:

[detached HEAD b96cc8fb864] Revert "nix-pin: add `passthru.updateScript`"
 1 file changed, 13 deletions(-)
[detached HEAD 94d92fd046f] Revert "nix-pin: 0.1.2 -> 0.2.2"
 1 file changed, 9 insertions(+), 8 deletions(-)

**Comment by timbertson** [comment]
Ah, damn, thanks for reverting. I thought it would be safe, but it looks like you can only use import-from-derivation behind a function (not as a passthru attribute). Fixed version: https://github.com/NixOS/nixpkgs/pull/39355

**Pull request merged**